cleanlab/datalab/internal/issue_manager/duplicate.py

```diff
@@ -79,7 +80,11 @@ def find_issues(
                     "If a knn_graph is not provided, features must be provided to fit a new knn."
                 )
             if self.metric is None:
-                self.metric = "cosine" if features.shape[1] > 3 else "euclidean"
+                self.metric = (
+                    "cosine"
+                    if features.shape[1] > 3
+                    else "euclidean" if features.shape[0] > 100 else euclidean
+                )
```

Comment:
this code is duplicated in multiple methods, should probably all be an imported helper method?
eg. `self.metric = get_knn_metric(features)`

Reply:
Thanks for pointing this out! This duplication comes from outlier.py ([see diff](https://github.com/cleanlab/cleanlab/pull/1113/files#diff-2831e7e2853b669892b9ed7c43245319d8b116093d2813d1a2c0b3b121131edfL429-L431)). The broader refactor you're suggesting would be valuable and aligns with what we want in #889. However, to avoid creeping elegance, let's handle that separately and keep this PR focused on the immediate bug fix (a numerically unstable algorithm for computing the Euclidean distance is affecting tests for all-identical examples). The docstrings ([primarily this one](https://github.com/cleanlab/cleanlab/pull/1113/files#diff-2831e7e2853b669892b9ed7c43245319d8b116093d2813d1a2c0b3b121131edfR64-R73)) address the current parameter choices; I'll ensure they're clear. We can revisit the refactor in a future PR.

Reply:
This PR has turned into a bit of [shotgun surgery](https://refactoring.guru/smells/shotgun-surgery). I suggest we handle the refactor separately to keep things manageable.

Reply:
at least put a TODO or GH issue? How is anybody going to remember to factor out the helper method later on?